docs/hooks/useDataQuery.md

```diff
@@ -134,3 +179,126 @@ export const IndicatorList = () => {
     )
 }
 ```
+
+### Multiple resources in one Query
+
+```tsx
+import React from 'react'
+import { useDataQuery, PaginatedQueryResult, Pager, PaginatedData } from '@dhis2/app-runtime'
+import { CircularLoader } from '@dhis2/ui'
+
+
+const query = {
+    dataElements: {
+        resource: "dataElements",
+        params: ({ dataElementsPage }: { dataElementsPage?: number }) => ({
+            fields: "id,displayName",
+            pageSize: 10,
+            page: dataElementsPage,
+        }),
+    },
+    dataSets: {
+        resource: "dataSets",
+        params: {
+            fields: "id,displayName",
+            paging: false,
+        },
+    },
+    indicators: {
+        resource: "indicators",
+        params: ({ indicatorsPage }: { indicatorsPage?: number }) => ({
+            fields: "id,displayName",
+            pageSize: 10,
+            page: indicatorsPage,
+        }),
+    },
+} as const;
+
+type QueryResult = {
+    dataElements: PaginatedData<{
+        //can wrap your data-type in utility type PaginatedData
+        dataElements: {
+            id: string;
+            displayName: string;
+        }[];
+    }>;
+    dataSets: {
+        // no pagination
+        dataSets: {
+            id: string;
+            displayName: string;
+        }[];
+    };
+    indicators: {
+        pager: Pager; // or you can specifiy the pager manually
+        indicators: {
+            id: string;
+            displayName: string;
+        }[];
+    };
+};
+export const Component = () => {
+    const { error, data, refetch } = useDataQuery<QueryResult>(query);
+
+    // keep previous data, so list does not disappear when refetching/fetching new page 
+    const prevData = React.useRef(data)
+    const stableData = data || prevData.current
+    React.useEffect(() => {
+        if(data) {
+            prevData.current = data
+        }
+    }, [data])
```

Comment:
Wasn't really sure if I should include this in the example. I do think we should either add an option to `useQuery` for `react-query`'s `keepPreviousData`. Or preferably just have an `reactQueryOptions` that we can pass directly... But I do understand the argument if we want to keep the APIs separate, to not rely on `react-query` if we were to replace it...

However I do think an example like this would be bad without the functionality of "keeping" old data when using pagination - especially since all lists would flicker/be hidden regardless of what "pager" updated, because `data` will be undefined when fetching... And since it's out of scope for this PR to have an option like that, I opted to include this pretty simple workaround in the example. And we can remove it/update it we decide to implement an option for this.